modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridCachePartitionsStateValidationTest.java

```diff
@@ -323,4 +330,136 @@ public void waitUntilAllSingleMessagesAreSent() throws IgniteCheckedException {
             allSingleMessagesSent.get();
         }
     }
+
+    /**
+     *  Test that partitions state validation works correctly and show partition size always (GG-24471):
+     *  1. Only partition counters are inconsistent.
+     *
+     * @throws Exception If failed.
+     */
+
+    /** Start three-nodes grid: first node (coordinator) with LogListener and two nodes without*/
+    private IgniteEx startGridWithFirsLoggerNode() throws Exception {
+        IgniteEx ignite = startGrid(0);
+            startGrid(1);
+            startGrid(2);
+            ignite.cluster().active(true);
+            awaitPartitionMapExchange();
+
+        // Populate cache to increment update counters.
+        for (int i = 0; i < 1000; i++)
+            ignite.cache(CACHE_NAME).put(i, i);
+
+        return ignite;
+    }
+
+    @Test
+    /**
+     *  Test that partitions state validation works correctly and show partition size always (GG-24471):
+     *  1. Only partition counters are inconsistent.
+     */
+    public void testValidationIfPartitionCountersAreInconsistentWithSize() throws Exception {
+        testLogger = new ListeningTestLogger();
+
+        LogListener lsnr = LogListener.matches("Partitions update counters are inconsistent for ").build();
+
+        testLogger.registerListener(lsnr);
+
+        try {
+            IgniteEx ignite = startGridWithFirsLoggerNode();
+
+            ignite.cachex(CACHE_NAME).context().topology().localPartitions().get(0).updateCounter(100500L);
+
+            // Modify update counter for some partition.
+
+            // Trigger exchange.
+            startGrid(3);
+
+            awaitPartitionMapExchange();
+
+            // Nothing should happen (just log error message) and we're still able to put data to corrupted cache.
+            ignite.cache(CACHE_NAME).put(0, 0);
+
+            assertTrue("Counters inconsistent message not found", lsnr.check());
+        }
+        finally {
```

Comment:
afterTest() is called by junit framework after each test. So try-finally construction can be just removed.